exporter/redis.go

```diff
@@ -711,27 +713,34 @@ func scanForKeys(c redis.Conn, pattern string) ([]string, error) {
 			return keys, fmt.Errorf("invalid response from SCAN for pattern: %s", pattern)
 		}
 
-		k, _ := redis.Strings(arr[1], nil)
+		k, err := redis.Strings(arr[1], nil)
+		if err != nil {
+			log.Debugln(err)
+		}
 		keys = append(keys, k...)
 
-		if iter, _ = redis.Int(arr[0], nil); iter == 0 {
+		if iter, err = redis.Int(arr[0], nil); iter == 0 {
```

Comment:
I think this change is not need nd you should remove 725-727\. you can ignore err here, iter will be zero even if an error is returned and there's no value in logging err, we can simply break.